File: Automation/001-AutomationFramework/C#/AutomationFramework/Common/CommonCore/Tests/TestStepInfo.cs
namespace CommonCore.Tests
{
    public class TestStepInfo : IStepInfo
    {
        #region Fields and Properties
        public string Description { get; set; }
        public bool SkipStep { get; set; }
        public Priority Level { get; set; }
        public bool SkipStepOnFailure { get; set; }
        public bool FailsIteration { get; set; }
        public DateTime StartTime { get; set; }
        public DateTime EndTime { get; set; }
        public TimeSpan Duration { get; set; }
        public Exception StepException { get; set; }
        public bool IsMandatory { get; set; }
        TestState Status { get; set; }
        TestState IStepInfo.Status { get; set; }
        #endregion

        public TestStepInfo()
        {
            StepInit();
        }

        public TestStepInfo(string description)
        {
            StepInit();
            Description = description;
        }

        public TestStepInfo(string description, Priority level)
        {
            StepInit();
            Description = description;
            Level = level;
        }

        public TestStepInfo(string description, bool skipStepOnFailure)
        {
            StepInit();
            Description = description;
            SkipStepOnFailure = skipStepOnFailure;
        }

        public TestStepInfo(string description, bool skipStepOnFailure, Priority level)
        {
            StepInit();
            Description = description;
            SkipStepOnFailure = skipStepOnFailure;
            Level = level;
        }

        public TestStepInfo(string description, bool skipStepOnFailure, Priority level, bool isMandatory = false)
        {
            StepInit();
            Description = description;
            SkipStepOnFailure = skipStepOnFailure;
            Level = level;
            IsMandatory = isMandatory;
        }

        public TestStepInfo(string description, bool skipStepOnFailure, bool skipStep)
        {
            StepInit();
            Description = description;
            SkipStepOnFailure = skipStepOnFailure;
            SkipStep = skipStep;
        }

        public TestStepInfo(string description, bool skipStepOnFailure, bool skipStep, Priority level)
        {
            StepInit();
            Description = description;
            SkipStepOnFailure = skipStepOnFailure;
            SkipStep = skipStep;
            Level = level;
        }

        protected void StepInit()
        {
            Description = "Description not defined";
            SkipStep = false;
            SkipStepOnFailure = true;
            FailsIteration = true;
            Level = Priority.High;
            Status = TestState.Unknown;
        }
    }
}

File: Automation/Woolsocks/TestAutomationFramework/src/main/java/common/core/tests/TestStepInfo.java
package common.core.tests;

import java.time.Duration;
import java.time.LocalDateTime;

public class TestStepInfo {
    // region Fields and Properties
    public String description;
    public boolean skipStep;
    public Priority level;
    public boolean skipStepOnFailure;
    public boolean failsIteration;
    public LocalDateTime startTime;
    public LocalDateTime endTime;
    public Duration duration;
    public Exception stepException;
    public boolean isMandatory;
    public TestState status;
    public String gherkinKeyword;
    // endregion

    public TestStepInfo() {
        stepInit();
    }

    public TestStepInfo(String description) {
        stepInit();
        this.description = description;
    }

    public TestStepInfo(String description, Priority level) {
        stepInit();
        this.description = description;
        this.level = level;
    }

    public TestStepInfo(String description, boolean skipStepOnFailure) {
        stepInit();
        this.description = description;
        this.skipStepOnFailure = skipStepOnFailure;
    }

    public TestStepInfo(String description, boolean skipStepOnFailure, Priority level) {
        stepInit();
        this.description = description;
        this.skipStepOnFailure = skipStepOnFailure;
        this.level = level;
    }

    public TestStepInfo(String description, boolean skipStepOnFailure, Priority level, boolean isMandatory) {
        stepInit();
        this.description = description;
        this.skipStepOnFailure = skipStepOnFailure;
        this.level = level;
        this.isMandatory = isMandatory;
    }

    public TestStepInfo(String description, boolean skipStepOnFailure, boolean skipStep) {
        stepInit();
        this.description = description;
        this.skipStepOnFailure = skipStepOnFailure;
        this.skipStep = skipStep;
    }

    public TestStepInfo(String description, boolean skipStepOnFailure, boolean skipStep, Priority level) {
        stepInit();
        this.description = description;
        this.skipStepOnFailure = skipStepOnFailure;
        this.skipStep = skipStep;
        this.level = level;
    }

    protected void stepInit() {
        description = "Description not defined";
        skipStep = false;
        skipStepOnFailure = true;
        failsIteration = true;
        level = Priority.HIGH;
        status = TestState.UNKNOWN;
    }
}
